awslabs/aws-crt-nodejs
# Typescript upgrade

* Update typescript to latest
* Update typedoc to almost-latest MV
* Update typedoc config to account for all the breaking changes
* Swap typedoc plugins to modern ones that still work and do approximately what we had plugins doing before.
* Rework how events are documented
* Move some private browser mqtt connection functionality to the bottom of the class definition
* Move some problematic type definitions that used CrtError (which is defined in browser/native) down to browser/native
* Expose checksums as an export
* Fix some messy type errors due to stricter checking in 4.x
* Update sample package.jsons

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
